packages/next/src/server/app-render/app-render.tsx

```diff
@@ -534,6 +600,11 @@ async function renderToHTMLOrFlightImpl(
     silenceLogger: silenceStaticGenerationErrors,
   })
 
+  let serverComponentsDidPostpone = false
+  const serverComponentsPostponeHandler = (_reason: unknown) => {
+    serverComponentsDidPostpone = true
+  }
+
```

Comment:
Not actually being used yet but this is going to help us distinguish the following cases

1. SSR or RSC postpone that was caught and not rethrown
2. SSR postpone handled by React
3. RSC postpone handled by React

previously we could detect postpones in SSR by virtue of the prerender rerturning some postponed state but the other cases appear identical. Since we need to support case 3 we need to distinguish this and we are going to use the information about whether Next initiated a Postpone paired with the whether React handled that postpone to tell these cases apart.

Side note: with the use of onPostpone we can get much more granular with our warning about catching postpones. Right now we'd not warn if the render still postponed even if your code did handle catch and hide one of the Next postpones. Ideally we'd warn if a single postpone was intercepted regardless of whether the render ultimately still postponed. I'll leave that exercise for a later time though